src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperators.vb

```diff
@@ -126,9 +140,34 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return left
         End Function
 
-        Private Function RewriteBinaryOperatorSimple(node As BoundBinaryOperator) As BoundNode
+        Private Shared Function GetLeftOperand(binary As BoundBinaryOperator, ByRef optimizeForConditionalBranch As Boolean) As BoundExpression
+            If optimizeForConditionalBranch AndAlso (binary.OperatorKind And BinaryOperatorKind.OpMask) <> BinaryOperatorKind.OrElse Then
+                Debug.Assert((binary.OperatorKind And BinaryOperatorKind.OpMask) = BinaryOperatorKind.AndAlso)
+                ' If left operand is evaluated to Null, three-valued Boolean logic dictates that the right operand of AndAlso
+                ' should still be evaluated. So, we cannot simply snap the left operand to Boolean.
+                optimizeForConditionalBranch = False
+            End If
+
+            Dim child As BoundExpression = binary.Left
+
+            While child.Kind = BoundKind.Parenthesized
+                child = DirectCast(child, BoundParenthesized).Expression
+            End While
+
+            Return child
+        End Function
+
+        Private Shared Function GetRightOperand(binary As BoundBinaryOperator, adjustIfOptimizableForConditionalBranch As Boolean) As BoundExpression
+            If adjustIfOptimizableForConditionalBranch Then
+                Return LocalRewriter.AdjustIfOptimizableForConditionalBranch(binary.Right, Nothing)
```

Comment:
>LocalRewriter. [](start = 23, length = 14)

Is the type qualifier needed?

Reply:
> Is the type qualifier needed?

If it is not, does its usage violate some policy?

---
In reply to: [328789857](https://github.com/dotnet/roslyn/pull/38802#discussion_r328789857) [](ancestors = 328789857)

Reply:
nit: I was thrown off by this too. Consider removing.

---
In reply to: [328821636](https://github.com/dotnet/roslyn/pull/38802#discussion_r328821636) [](ancestors = 328821636,328789857)

Reply:
>  I was thrown off by this too.

Could you please explain what confused you?

---
In reply to: [328859772](https://github.com/dotnet/roslyn/pull/38802#discussion_r328859772) [](ancestors = 328859772,328821636,328789857)

Reply:
I thought I had seen that method in LocalRewriter and so went to check where it was. When we qualify, I assume that the method is somewhere else, even though I recognize it is not necessary ;-)

---
In reply to: [328863180](https://github.com/dotnet/roslyn/pull/38802#discussion_r328863180) [](ancestors = 328863180,328859772,328821636,328789857)

Reply:
> I thought I had seen that method in LocalRewriter and so went to check where it was. When we qualify, I assume that the method is somewhere else, even though I recognize it is not necessary ;-)

So, there was an actual benefit for the purpose of code review. You saw the call and went to see what the method actually does. Seems like a huge plus to me.  :-) 

---
In reply to: [328863963](https://github.com/dotnet/roslyn/pull/38802#discussion_r328863963) [](ancestors = 328863963,328863180,328859772,328821636,328789857)

Reply:
I didn't explain clearly. I was wondering "how does this relate to the method I saw earlier?" (ie. is it the same one, or a different one). The qualifier is what made me doubt. It's a nit. When the code is loaded in VS that won't be a problem (go to def...).

---
In reply to: [328866218](https://github.com/dotnet/roslyn/pull/38802#discussion_r328866218) [](ancestors = 328866218,328863963,328863180,328859772,328821636,328789857)

Reply:
> Is the type qualifier needed?

> didn't explain clearly. I was wondering "how does this relate to the method I saw earlier?" (ie. is it the same one, or a different one). The qualifier is what made me doubt. It's a nit. When the code is loaded in VS that won't be a problem (go to def...).

The type qualifier is need here because VB is case-insensitive and parameter takes over otherwise.

---
In reply to: [328867784](https://github.com/dotnet/roslyn/pull/38802#discussion_r328867784) [](ancestors = 328867784,328866218,328863963,328863180,328859772,328821636,328789857)